presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -390,6 +392,13 @@ public PlanOptimizers(
                         costCalculator,
                         ImmutableSet.of(new RemoveEmptyDelete()))); // Run RemoveEmptyDelete after table scan is removed by PickTableLayout/AddExchanges
 
+         // Dynamic Filter Rule should be followed later by Predicate Pushdown as pushing dynamic filters deep down is beneficial
+        builder.add(new IterativeOptimizer(
```

Comment:
merge this commit with `Add ApplyDynamicFilters`?

Reply:
This should be guarder with a future flag as well.

Reply:
Done !!

Reply:
Rule interface has isEnabled() which checks for flag and optimizer ignores rule based on that.